src/httpProblemDetails.ts

```diff
@@ -0,0 +1,39 @@
+import { HttpProblemDetailsDefinition } from './httpProblemDetailsDefinition';
+
+const invalidDetailName = ['type', 'status', 'title'];
+
+export class HttpProblemDetails {
+  public readonly type: string;
+  public readonly title: string;
+  public readonly status: number;
+  private additionalDetails: Map<string, any>;
+
+  constructor({ type, title, status }: HttpProblemDetailsDefinition) {
+    this.type = type;
```

Comment:
According to RFC, detail has a default type `about:blank` in a pretty common situation:
```
when used as a problem type, indicates that the problem has no additional semantics beyond that of the HTTP status code.
```

This means that `type` in constructor should be optional. But not only that. Acccording to RFC:

```
   When "about:blank" is used, the title SHOULD be the same as the
   recommended HTTP status phrase for that code (e.g., "Not Found" for
   404, and so on), although it MAY be localized to suit client
   preferences (expressed with the Accept-Language request header).
```

So maybe we need to find a way to infer title based on status given?

Reply:
This first part I think is essential; the second is optional and probably for later stages of development